app/models/guest_device.rb

```diff
@@ -10,4 +11,7 @@ class GuestDevice < ApplicationRecord
 
   has_one :network, :foreign_key => "device_id", :dependent => :destroy, :inverse_of => :guest_device
   has_many :miq_scsi_targets, :dependent => :destroy
+
+  has_many :firmwares, :dependent => :destroy
+  has_many :child_devices, -> { where(:parent_device_id => ids) }, :foreign_key => "parent_device_id", :class_name => "GuestDevice", :dependent => :destroy
```

Comment:
@blomquisg I made the changes you suggested, and they seemed to work at first, but after I restarted the evm, one of the ports associated with the guest device was replaced by a duplicate of the other port in the table. For example, suppose we have guest device 1 that has port 1 and port 2. At first, the guest device and its associated ports are populated in the DB correctly. After a evm restart is performed, port 1 disappears from the table and there are now two duplicate port 2 entries with the same address and parent device. Any ideas why this is occurring? Does it have something to do with the save_inventory code? Including @juliancheal @Fryguy